_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,336 @@
+---
+id: efbfc7
+name: Auto-updating text content can be paused or stopped
+rule_type: atomic
+
+description: |
+  This rule checks that there are mechanisms to pause or stop the auto-updating of text content
+
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:2.2.2: # Pause, Stop, Hide (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+
+authors:
+  - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [text node][] that is [visible][] and a [child][] in the [flat tree][] of an HTML element, whose content changes through an action that was not initiated by the user, anytime after the [readiness][document readiness] of the document the [text node][] belongs to is equal to "complete".
+
+## Expectation 1
+
+For each test target a [mechanism][] is provided to pause or stop the updating of the content of the [text node][].
+
+**Note**: If there is more than one [text node][] with auto-updating content, a single [mechanism][] may be used to pause or stop updating all [text nodes][text node].
+
+## Expectation 2
+
+The [mechanism][] to pause or stop the auto-updating of the content of the [text node][] is [visible][], has an [accessible name][] that is not empty ("") nor only [ASCII whitespace][] , and is [included in the accessibility tree][].
+
+## Assumptions
+
+This rule assumes that the auto-updating of the content is not [essential][], which is listed as valid exception to SC 2.2.2.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.2.2: Pause, Stop, Hide](https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html)
+- [G186: Using a control in the Web page that stops moving, blinking, or auto-updating content](https://www.w3.org/WAI/WCAG21/Techniques/general/G186)
+- [F16: Failure of Success Criterion 2.2.2 due to including scrolling content where movement is not essential to the activity without also including a mechanism to pause and restart the content](https://www.w3.org/WAI/WCAG21/Techniques/failures/F16)
+
+## Test Cases
```

Comment:
Let us say I have an async fn that fetches data, in the meantime, the view says `loading`, then I also want to keep changing the text 
- `loading from skynet`
- `rebooting drives`
- `the sun is a big orange` etc., when the user is waiting

then finally after data get, I render

Is this considered auto updating text? In my opinion it is a status/ progress bar. 
All flight search websites do this today.

Reply:
I would perhaps argue that such auto-updating text is essential, which leaves me questioning the assumption that it isn't.

Reply:
I tend to agree with Kasper. I would say that status bar is essential auto-updating text.

Reply:
I agree that a status update bar would be considered essential auto-updating text.

SC 2.2.2 has this note: An animation that occurs as part of a preload phase or similar situation can be considered essential if interaction cannot occur during that phase for all users and if not indicating progress could confuse users or cause them to think that content was frozen or broken.

Reply:
As previously mentioned, I agree that this is an essential auto-update, therefore, as stated in the assumption and SC, there is no need to have a mechanism to control it.